packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/MrwSerializationTypeProvider.cs

```diff
@@ -156,30 +188,37 @@ protected override MethodProvider[] BuildMethods()
         /// <returns>An array of <see cref="CSharpType"/> types that the model implements.</returns>
         protected override CSharpType[] BuildImplements()
         {
-            int interfaceCount = _iJsonModelObjectInterface != null ? 2 : 1;
-            CSharpType[] interfaces = new CSharpType[interfaceCount];
-            interfaces[0] = _iJsonModelTInterface;
-
+            List<CSharpType> interfaces = [_iJsonModelTInterface];
+            CSharpType? inheritedType = _model.Inherits;
+            if (inheritedType != null)
+            {
+                interfaces.Add(inheritedType);
+            }
             if (_iJsonModelObjectInterface != null)
             {
-                interfaces[1] = _iJsonModelObjectInterface;
+                interfaces.Add(_iJsonModelObjectInterface);
             }
 
-            return interfaces;
+            return interfaces.ToArray();
         }
 
         /// <summary>
         /// Builds the JSON serialization write method for the model.
         /// </summary>
         internal MethodProvider BuildJsonModelWriteMethod()
         {
+            var methodName = $"{nameof(IJsonModel<object>.Write)}Core";
             ParameterProvider utf8JsonWriterParameter = new("writer", $"The JSON writer.", typeof(Utf8JsonWriter));
-            // void IJsonModel<T>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+            MethodSignatureModifiers modifiers = MethodSignatureModifiers.Protected | MethodSignatureModifiers.Virtual;
+            if (_model.Inherits != null)
```

Comment:
I think this check is not enough or precise enough - in the future we definitely will have customization code support - when that happens, the inherits could come from customization code, therefore we need to check the condition of writing "WriteCore" method more precisely.

At least it should be something like
```suggestion
            if (_model.Inherits is {IsFrameworkType: false, Implementation: ModelProvider })
```
to ensure the base type is a model generated by us.
Potentially we might need to check if the base type has this virtual method of `WriteCore` before we could write an override method, but we could do the simple way for now and improve in the future.